easystepper/easystepper.go

```diff
@@ -2,28 +2,76 @@
 package easystepper // import "tinygo.org/x/drivers/easystepper"
 
 import (
+	"errors"
 	"machine"
 	"time"
 )
 
+// StepMode determines the coil sequence used to perform a single step
+type StepMode uint8
+
+// Valid values for StepMode
+const (
+	// ModeFour uses a 'four step' coil sequence (12-23-34-41). This is the default (zero-value) mode
+	ModeFour StepMode = iota
+	// ModeEight uses an 'eight step' coil sequence (1-12-2-23-3-34-4-41)
+	ModeEight
+)
+
+// stepCount is a helper function to return the number of steps in a StepMode sequence
+func (sm StepMode) stepCount() uint {
+	switch sm {
+	default:
+		fallthrough
+	case ModeFour:
+		return 4
+	case ModeEight:
+		return 8
+	}
+}
+
+// DeviceConfig contains the configuration data for a single easystepper driver
+type DeviceConfig struct {
+	// Pin1 ... Pin4 determines the pins to configure and use for the device
+	Pin1, Pin2, Pin3, Pin4 machine.Pin
+	// StepCount is the number of steps required to perform a full revolution of the stepper motor
+	StepCount uint
```

Comment:
It makes sense  to set StepCount and RPM to uint types since it makes no physical sense for it to be a negative number, but this can hide user input errors like bad calculations, i.e: calculating RPM from radians per second or the like. It's usually a good idea to use the `int` type for all integer inputs except in specific cases. This is my personal opinion and comes from experience using other Go APIs I've seen in the wild.

Reply:
Hmm, that's a good point. I'm also thinking it should maybe be explicit in size too, i.e. `int32`? I'm not sure of the rules for a 'machine word' in TinyGo, but if `int` & `uint` are only 8-bits on 8-bit MCUs (e.g. AVR) then it should be explicit to provide for a larger range than 127!